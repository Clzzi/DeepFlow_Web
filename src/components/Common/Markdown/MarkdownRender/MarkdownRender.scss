.MarkdownRender {
	p,
	div,
	pre {
		font-size: var(--font-contents) !important;
		line-height: 2.2rem;
	}

	h1,
	h2,
	h3,
	h4,
	h5,
	h6 {
		color: var(--contrast);
		margin-bottom: 0.25rem;
	}
}
