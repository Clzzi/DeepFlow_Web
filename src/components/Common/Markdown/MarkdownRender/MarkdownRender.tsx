import { CSSProperties } from 'react';
import ReactMarkdown from 'react-markdown';
import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';
import gfm from 'remark-gfm';
import BlockQuote from './BlockQuote';
import CodeBlock from './CodeBlock';
import Image from './Image';

const style = require('./MarkdownRender.scss');
const cx: ClassNamesFn = classNames.bind(style);

export interface MarkdownRenderProps {
	contents: string;
	style?: CSSProperties;
}

const MarkdownRender = ({
	contents,
	style,
}: MarkdownRenderProps): JSX.Element => {
	return (
		<div
			className={cx('MarkdownRender')}
			style={style}
		>
			<ReactMarkdown
				className={cx('MarkdownRender-Render')}
				source={contents}
				plugins={[gfm]}
				escapeHtml={false}
				renderers={{
					code: CodeBlock,
					blockquote: BlockQuote,
					image: Image,
				}}
			/>
		</div>
	);
};

export default MarkdownRender;
