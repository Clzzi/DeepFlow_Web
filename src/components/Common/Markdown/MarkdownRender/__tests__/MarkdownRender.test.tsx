import { render } from '@testing-library/react';
import MarkdownRender, { MarkdownRenderProps } from '../MarkdownRender';

describe('MarkdownRender', () => {
  const initialSetup = (props: Partial<MarkdownRenderProps> = {}) => {
    const initialProps: MarkdownRenderProps = {
      contents: '',
      ...props,
    };

    return render(<MarkdownRender {...initialProps} />);
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });
});