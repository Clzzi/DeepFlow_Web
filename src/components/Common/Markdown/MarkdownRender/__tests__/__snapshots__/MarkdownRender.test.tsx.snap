// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownRender render correctly 1`] = `
<div>
  <div
    class="MarkdownRender"
  >
    <div
      class="MarkdownRender-Render"
    />
  </div>
</div>
`;
